addons/web/static/src/js/views/basic/basic_model.js

```diff
@@ -877,6 +877,10 @@ var BasicModel = AbstractModel.extend({
             parent._changes = _.filter(parent._changes, function (change) {
                 return change.id !== elementID;
             });
+            // The record has been deleted from the _changes.
+            // So we won't be passing into the logic of _applyX2ManyOperations anymore
+            // implying that we have to cancel out the effects of a CREATE command here
+            parent.limit--;
```

Comment:
The limit is increased only when it is reached (in the CREATE op handler). Here, you always decrease it, so it's not ok.